contracts/src/v0.1/Prepayment.sol

```diff
@@ -440,24 +444,28 @@ contract Prepayment is Ownable, IPrepayment, ITypeAndVersion {
         uint256 protocolFee = (amount * sProtocolFeeRatio) / 100;
         uint256 operatorFee = amount - burnFee - protocolFee;
 
-        account.chargeFee(
-            burnFee,
-            operatorFee,
-            operatorFeeRecipient,
-            protocolFee,
-            sProtocolFeeRecipient
-        );
+        account.chargeFee(burnFee, operatorFee, protocolFee, sProtocolFeeRecipient);
 
         emit AccountBalanceDecreased(accId, balance, balance - amount, burnFee);
```

Comment:
This related to the first comment.

Reply:
@KelvinThai This is a bit off-topic but let me try to get to the point. I think we might have to think of some fund locking mechanism, where the steps would be as follows:

1. request something (vrf, rr)
2. lock funds to be able to pay for fulfillment (locked funds cannot be withdrawn, only when request is canceled)
3. fulfill first, fulfill second and so on (fulfillments are taken separately by each reporter - not just last one)
4. each fulfillment decreases locked funds and sends $KLAY to reporter's account and emits `BalanceDecreased` event
5. the last fulfillment aggregates submissions and execute consumer's fulfill function

It is a bit different from what is happening now, but I think it would be easier to implement, easier to read and we don't have to punish reporters by waiting until the last reporter submits if at all.

Let me know what you think :)

Reply:
Good idea. but not easy to execute.
1. we need to know in advance how much to lock.
2. Consumers have not received data but still, deduct the balance.
3. we will run into problems I raised before, what if the report alr registered but they dont run at all.
Let's think of the best method to solve all in one.